gap/base/recognition.gi

```diff
@@ -528,7 +528,11 @@ InstallGlobalFunction( RecogniseGeneric,
               counter,").");
         fi;
         Add(depth,'F');
-        Assert(2, ForAll(GeneratorsOfGroup(H), x->ValidateHomomInput(ri, x)));
+        if ForAny(GeneratorsOfGroup(H), x->not ValidateHomomInput(ri, x)) then
+            # Our group fails to contain some of the generators of H!
+            return fail;
+        fi;
```

Comment:
Hrm, I am not completely sure if I believe that this is the correct thing to do at this point. In any case, I think it's closely related to the mandarin work. I'll have to think about this for a little bit (but not right now).

That said, the worst this can do is break something that worked before; the best is that it prevents some incorrect results; so overall, even if it needs to be done slightly differently in the end, I guess it won't hurt as long as the tests keep passing :-)

Reply:
This certainly should be fixed with mandarins, we shouldn't really get to this place -- it means (I think) we've decided on a homomorphism where the generators of the group aren't valid inputs.

Might be worth thinking about if the set of mandarins should always include the generators. On one hand they are "low quality", on the other hand any recognised group really better include the generators the user gave :)